### DISCUSSION TITLE
Set Preview Mode cookies by path

### DISCUSSION THREAD

**Discussion by lucasassisrosa** [vercel/next.js#38311] [Help]
Using Next.js Preview Mode feature:

```
const handler: NextApiHandler = (req, res) => {
 ...
  const post = await getPreviewPostBySlug(slug)

  // If the slug doesn't exist prevent preview mode from being enabled
  if (!post) {
    return res.status(401).json({ message: 'Invalid slug' })
  }

  // Enable Preview Mode by setting the cookies
  res.setPreviewData({});
  ...
};
export default handler;
```

Whenever I call `res.setPreviewData`, any other API Requests made for other `/pages/...` will have the `COOKIE_NAME_PRERENDER_BYPASS` and `COOKIE_NAME_PRERENDER_DATA` cookies included. Checking the code, I see that the cookies are being set for path `"/"`:

https://github.com/vercel/next.js/blob/9d22da476b5439f6a925f9e4c2758a499a610e7b/packages/next/server/api-utils/node.ts#L513
https://github.com/vercel/next.js/blob/9d22da476b5439f6a925f9e4c2758a499a610e7b/packages/next/server/api-utils/node.ts#L522
 
Can we update the logic to accept a "cookies path" parameter? Sometimes we want to enable preview mode for a page but not all the other ones.

I'm using 

```
“next”:"12.1.0",
"react": "17.0.1",
"react-dom": "17.0.1",
```

**Comment by mikalai-t** [comment]
It seems this has been released in 12.2.5
